Solr textContainsPhrase & Solr Client Tokenizer Alignment

criminosis opened the pull request:
-----

Thank you for contributing to JanusGraph!

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [X] Is there an issue associated with this PR? Is it referenced in the commit message?
- [X] Does your PR body contain #xyz where xyz is the issue number you are trying to resolve?
- [X] Has your PR been rebased against the latest commit within the target branch (typically `master`)?
- [X] Is your initial contribution a single, squashed commit?

### For code changes:
- [X] Have you written and/or updated unit tests to verify your changes?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](https://www.apache.org/legal/resolved.html#category-a)?
- [ ] If applicable, have you updated the LICENSE.txt file, including the main LICENSE.txt file in the root of this repository?
- [ ] If applicable, have you updated the NOTICE.txt file, including the main NOTICE.txt file found in the root of this repository?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

Fixes https://github.com/JanusGraph/janusgraph/issues/4165
Closes https://github.com/JanusGraph/janusgraph/issues/4164

li-boxuan commented:
@VladimirBogomolov Would you like to review?

li-boxuan reviewed (commented):
Thank you for the fix and the new pushdown rule! I have one question about the tokenizer fix:

In your issue description https://github.com/JanusGraph/janusgraph/issues/4165 you mentioned that you witnessed the bug when migrating from Elasticsearch to Solr. Is it possible to write a test in `JanusGraphIndexTest` such that Elasticsearch backend would pass, and Solr backend wouldn't without your fix? I understand you added a new unit test for Solr backend, but it would be great if we could have a test to ensure the consistent behaviour across index backends.

criminosis commented:
I believe running the test class I modified without the modified Solr client should demonstrate it. The test class I modified appeared to be executed upon all providers if I'm not mistaken.

li-boxuan commented:
> The test class I modified appeared to be executed upon all providers if I'm not mistaken.

Oh yeah you are right

criminosis commented:
Restoring the original client will opt it out of the containsPhrase testing based on the support flags evaluated by the supports method. But I believe it should demonstrate the breakage with regards to the textContains predicate.

If not I woefully missed up in my preliminary investigation 😅

li-boxuan reviewed (approved):
Oh gosh sorry I completely forgot about this PR. If there's no other review, I'll merge this PR in a week following lazy merge consensus.

li-boxuan commented:
@criminosis Sorry I completely forgot about this PR (again)... Do you mind rebasing this PR against latest master branch to retrigger the CI?

criminosis commented:
Done @li-boxuan 👍
